code/modules/research/xenoarchaeology/tools/tools_coresampler.dm

```diff
@@ -50,16 +50,16 @@
 				icon_state = "sampler-full"
 
 				//put in a rock sliver
-				var/obj/item/weapon/rocksliver/R = new(src)
-				R.geological_data = geo_data
+				extracted = new(src)
```

Comment:
Hard deletes if, let's say, an explosion destroys this object before the person ejects the sample.

Create a Destroy() for this object and clear `extracted` if it exists

Reply:
done